### DISCUSSION TITLE
Migrate Non-Docker to Docker

### DISCUSSION THREAD

**Discussion by iTABHayden** [immense/Remotely#545] [Q&A]
Hi there,

Since there won't be a new non-docker version of Remotely, how do we migrate our old, on docker (legacy?) installations to Docker? For us ones with a setup database and many endpoints this would be useful.

Cheers.

**Comment by goftari** [comment]
Hi
You may find the instruction on README file:
https://github.com/immense/Remotely#after-installation 
Data for Remotely will be saved in /var/www/remotely/ within two files: appsettings.json and Remotely.db.
These files will persist through teardown and setup of new Remotely containers.
If upgrading from a non-Docker version of Remotely, overwrite these files with the ones from your previous installation.

**Reply by avl55-labs** [reply]
@goftari Hi! This way won't work for me (non-docker to docker upgrade). Docker fresh install won't see database, asking for register of the new admin user after start. May be i do it wrong and Remotely.db shoud be placed some where else than /var/www/remotely/ Can you help ?

**Comment by shastybsd** [comment]
I need to use this path: 

/var/lib/docker/overlay2/e7de516b6cf9f595e1bbcdc276db4e6d9ce1340c6dc17b1507f51cffc6425ee1/diff/app/Remotely.db

**Comment by Rhyzar1** [comment]
I am having the same issue migrating from non-docker to the new docker image model.
When deploying the image I used "-v /var/www/remotely:/remotely-data" as stated in the quickstart.
I see the container is mapping /var/www/remotely to /remotely-data
I replaced the files in /var/www/remotely from my old server directory.
...but as far as I can tell, Remotely is not using the appsettings.json and Remotely.db from /var/www/remotely
I believe that it is using the files from /var/lib/docker/overlay2/fa6c3f5e72ff33b08aee1f553a002be0b2d33579076c33fb592375a8364bc8cd/diff/app/
I can replace these files in here, but when updating the remotely docker image (docker rm remotely) it will delete that directory and the files as well.

Is the mapping of -v /var/www/remotely:/remotely-data actually correct or even working?
inside the /var/lib/docker/overlay2/fa6c3f5e72ff33b08aee1f553a002be0b2d33579076c33fb592375a8364bc8cd/diff/app/ directory I see:
lrwxrwxrwx 1 root root      31 Mar  6 03:01 appsettings.json -> /remotely-data/appsettings.json
-rw-r--r-- 1 root root    4096 Mar  6 03:01 Remotely.db
-rw-r--r-- 1 root root   32768 Mar  6 03:01 Remotely.db-shm
-rw-r--r-- 1 root root 1318432 Mar  6 03:01 Remotely.db-wal

I am really new to docker and I am not understanding how this is supposed to use the files in /var/www/remotely.

ANY help would be appreciated.

**Comment by Rhyzar1** [comment]
OK, so digging deeper into this.
I went into the container by using "sudo docker exec -it remotely /bin/bash"
I see the symbolic link in /app for "appsettings.json -> /remotely-data/appsettings.json", but there was no symbolic link for Remotely.db.
I deleted the Remotely.db from /app and created a symbolic link to the Remotely.db in /remotely-data: "ln -s /remotely-data/Remotely.db Remotely.db"
I stopped remotely: "sudo docker stop remotely" and started remotely: "sudo docker start remotely"
Now it looks like it is using the correct Remotely.db as I can log in and see all of my remote devices & scripts.

Now when pointing the reverse proxy over to this new instance of Remotely, none of the remote devices are coming online.

I am not sure what is going on or even what might be going wrong.

I am coming from version v2022.08.02.0428 I believe.....
